packages/base/src/3dview/mainview.tsx

```diff
@@ -442,58 +442,61 @@ export class MainView extends React.Component<IProps, IStates> {
                 }
               }
             });
-          } else if (
-            this._transformControls.mode === 'rotate' &&
-            this._pivot &&
-            this._matchingChild
-          ) {
-            // Retrieve Euler rotations for pivot and matching child
-            const pivotEuler = this._pivot.rotation;
-            const matchingChildEuler = this._matchingChild.rotation;
-
-            // Create a new Euler by adding rotations component-wise
-            const combinedEuler = new THREE.Euler(
-              pivotEuler.x * matchingChildEuler.x,
-              pivotEuler.y * matchingChildEuler.y,
-              pivotEuler.z * matchingChildEuler.z,
-              pivotEuler.order // Ensure we use the same rotation order
+        } else if (this._transformControls.mode === 'rotate' && this._pivot) {
+            // Retrieve the existing rotation from the shared model
+            const existingRotationArray = obj?.parameters?.Placement?.Axis;
+            const existingRotation = new THREE.Euler(
+                THREE.MathUtils.degToRad(existingRotationArray[0]),
+                THREE.MathUtils.degToRad(existingRotationArray[1]),
+                THREE.MathUtils.degToRad(existingRotationArray[2]),
```

Comment:
The Placement axis is not the Euler rotation angle. You could check the definition here https://wiki.freecad.org/Placement